core/freq_params_traffic/traffic_shaping.go

```diff
@@ -0,0 +1,273 @@
+package freq_params_traffic
+
+import (
+	"fmt"
+	"math"
+	"runtime"
+	"sync/atomic"
+
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/alibaba/sentinel-golang/core/freq_params_traffic/cache"
+	"github.com/alibaba/sentinel-golang/util"
+)
+
+type TrafficShapingController interface {
+	PerformChecking(arg interface{}, acquireCount int64) *base.TokenResult
+
+	BoundParamIndex() int
+
+	BoundMetric() *ParamsMetric
+
+	BoundRule() *Rule
+}
+
+type baseTrafficShapingController struct {
+	r *Rule
+
+	res           string
+	metricType    MetricType
+	paramIndex    int
+	threshold     float64
+	specificItems map[interface{}]int64
+	durationInSec int64
+
+	metric *ParamsMetric
+}
+
+func newBaseTrafficShapingControllerWithMetric(r *Rule, metric *ParamsMetric) *baseTrafficShapingController {
+	size := int(math.Min(float64(ParamsMaxCapacity), float64(ParamsCapacityBase*r.DurationInSec)))
```

Comment:
How about making the cache size as an attribute of the param flow rule?

Reply:
Maybe an optional attribute make sense.